### PR TITLE
Flag to output the score of each sentence at inference + various scoring tools

### PR DESCRIPTION
**Pull request by vince62s**
The goal is to be able to rerank nbest hypothesis when combining the score of each translation with another score.
In the context of Noisy Channel Reranking we combine:
the score of each translation
the backward score with an inverse model
a language model score of each translation

New tools:
mbr_bleu: compute the best bleu amongst n_best hypotheses between themselves.
oracle_bleu: compute the best possible BLEU against a reference and pick the best in n_best
oracle_comet: same with unbable/comet

### CODE REVIEW

File: onmt/translate/translator.py [hunk 1]
```diff
@@ -496,6 +499,10 @@ def _translate(
                 n_best_preds = [
                     " ".join(pred) for pred in trans.pred_sents[: self.n_best]
                 ]
+
+                n_best_scores = list(map(lambda x: x.item(),
+                                         trans.pred_scores[: self.n_best]))
```
Comment (francoishernandez) [inline onmt/translate/translator.py:504]: A list comprehension might be a bit more readable (and coherent with the operations above).
https://stackoverflow.com/questions/1247486/list-comprehension-vs-map

File: onmt/translate/translator.py [hunk 2]
```diff
@@ -515,7 +522,14 @@ def _translate(
                     n_best_preds = [transform.apply_reverse(x)
                                     for x in n_best_preds]
                 all_predictions += [n_best_preds]
-                self.out_file.write("\n".join(n_best_preds) + "\n")
+
+                out_all = list(map(lambda x: x[0] + "\t" + str(x[1]),
+                                   zip(n_best_preds, n_best_scores)))
```
Comment (francoishernandez) [inline onmt/translate/translator.py:527]: Same.

**Review by francoishernandez — COMMENTED** [review]
Did not review tools. Probably not critical.
Would be nice to document those a little bit though. Either in the README in the tools folder, or in the header of each script.

File: onmt/translate/translator.py (hunk 1 above)
Comment (vince62s) [reply onmt/translate/translator.py:504]: done

File: onmt/translate/translator.py (hunk 2 above)
Comment (vince62s) [reply onmt/translate/translator.py:527]: done

**Comment by francoishernandez** [comment]
Thanks, merging!

**Pull request merged**